haskell/test/Lorentz/Contracts/Test/Permit.hs

```diff
@@ -82,6 +89,65 @@ assertPermitCount contractAddr expectedCount =
       sum $
         permits <&> \userPermits -> length (upPermits userPermits)
 
+checkView
+  :: forall viewParam viewVal
+   . (IsoValue viewParam, IsoValue viewVal, Eq viewVal, Show viewVal)
+  => TAddress Parameter -> Text -> viewParam -> viewVal -> IntegrationalScenario
+checkView addr viewName viewParam expectedViewVal =
+  lExpectStorage @Storage addr $ \st -> do
+    uri <-
+      onNothing "Metadata bigmap did not contain a key with the empty string" $
+        fmap (decodeUtf8 @Text) . Map.lookup mempty . unBigMap $ sMetadata st
+
+    metadataJSONKey <-
+      onNothing ("Expected 'tezos-storage' uri, but found: " <> uri) $
+        Text.stripPrefix "tezos-storage:" uri
+
+    metadataJSONKeyMText <-
+      first
+        (\err -> CustomTestError $
+            "Expected '" <> metadataJSONKey <> "' to be a valid Michelson string, but it wasn't."
+            <> "\nReason: " <> err
+        ) $
+        mkMText metadataJSONKey
+
+    metadataJSONBytestring <-
+      onNothing ("Metadata bigmap did not contain the key: " <> metadataJSONKey) $
+        Map.lookup metadataJSONKeyMText . unBigMap $ sMetadata st
+
+    metadataJSON <-
+      first
+        (\err -> CustomTestError (toText err)) $
+        J.eitherDecode' @(TZ.Metadata (ToT Storage)) (BSL.fromStrict metadataJSONBytestring)
+
+    view_ <-
+      onNothing ("Metadata does not contain view with name" <> viewName) $
+        TZ.getView metadataJSON viewName
+
+    -- See note below.
+    unless (TZ.vPure view_) $
+      Left $ CustomTestError $ "Expected view '" <> viewName <> "' to be pure, but it isn't."
+
+    actualViewVal <-
+      first
+        (CustomTestError . pretty) $
+        -- NOTE: it's OK to use `dummyContractEnv` here because we now this
+        -- contract's views are pure (i.e. don't depend on the contract env).
+        -- But, in the general case, this is not safe.
+        TZ.interpretView @_ @_ @viewVal dummyContractEnv view_ st viewParam
+
+    when (actualViewVal /= expectedViewVal) $
+      Left $ CustomTestError $ unlines
+        [ "Expected: "
+        , show expectedViewVal
+        , "Got:"
+        , show actualViewVal
+        ]
+  where
+    onNothing :: Text -> Maybe a -> Either TestError a
```

Comment:
There is [`maybeToRight`](https://hackage.haskell.org/package/universum-1.7.1/docs/Universum-Monad-Either.html#v:maybeToRight), so this `onNothing` can be simplified a bit.

Reply:
Ah, forgot about that! I deleted `onNothing`